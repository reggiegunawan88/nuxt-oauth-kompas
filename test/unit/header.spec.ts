import { shallowMount } from "@vue/test-utils";
import Header from '~/components/header/header.vue'
describe('Header.vue test', () => {
  // mock route
  const $route = {
    query: {
      prevurl: '/'
    }
  }
  const wrapper = shallowMount(Header, {
    stubs: { fontAwesomeIcon: true },
    mocks: {
      $route,
    }
  })
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  test('header back button', () => {
    const button = wrapper.find('#btnBack')
    const fontawesomeicon = wrapper.find('#btnIcon')
    const textBtn = wrapper.find("#txtKembali")
    expect(button).toBeTruthy()
    expect(fontawesomeicon).toBeTruthy()
    expect(textBtn).toBeTruthy()
  });
  test('render image logo', () => {
    const img = wrapper.findAll('#logoKompas')
    expect(img.length).toBe(2)
  });
});