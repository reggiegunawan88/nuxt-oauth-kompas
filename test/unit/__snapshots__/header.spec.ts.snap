// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue test snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
